zcncore/smartcontract_executor_mobile.go

```diff
@@ -0,0 +1,33 @@
+//go:build mobile
+// +build mobile
+
+package zcncore
+
+import "encoding/json"
+
+type SmartContractExecutor interface {
+	// ExecuteSmartContract implements wrapper for smart contract function
+	ExecuteSmartContract(address, methodName string, input string, val uint64) error
+}
+
+func (t *Transaction) ExecuteSmartContract(address, methodName string, input string, val uint64) error {
+	err := t.createSmartContractTxn(address, methodName, json.RawMessage(input), val)
```

Comment:
Doing the json.RawMessage(input) won't fix this issue I said before, this is just dong a []byte() conversion, but the input data structure that 0chain expects is a json object, or a map[string]interface{} that could represents the json object. Please decode the input to a map[string]interface{}, then pass it to the `t.createSmartContractTxn`.

Reply:
https://go.dev/play/p/G8w3cO1DaPF

Reply:
yes, didn't know the json.RawMessage could be used for this. 👍